cnf-tests/testsuites/e2esuite/multinetworkpolicy/multinetworkpolicy_sriov.go

```diff
@@ -0,0 +1,819 @@
+						PodSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"pod": "b",
+							},
+						},
+					}},
+				}),
+				CreateInNamespace(nsX),
+			)
+
+			// IPv4
+			eventually30s(nsX_podB).Should(Reach(nsX_podA, ViaIPv4))
+			eventually30s(nsX_podC).ShouldNot(Reach(nsX_podA, ViaIPv4))
+			eventually30s(nsX_podB).Should(Reach(nsX_podC, ViaIPv4))
+
+			// IPv6
+			eventually30s(nsX_podB).Should(Reach(nsX_podA, ViaIPv6))
+			eventually30s(nsX_podC).ShouldNot(Reach(nsX_podA, ViaIPv6))
+			eventually30s(nsX_podB).Should(Reach(nsX_podC, ViaIPv6))
+		})
+	})
+
+})
+
+func cleanPodsInNamespaces(argNamespaces ...string) {
+	for _, ns := range argNamespaces {
+		namespaces.CleanPods(ns, client.Client)
+	}
+}
+
+func cleanMultiNetworkPoliciesFromNamespace(namespace string) {
+	err := client.Client.MultiNetworkPolicies(namespace).
+		DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	Eventually(func() int {
+		ret, err := client.Client.MultiNetworkPolicies(namespace).
+			List(context.Background(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		return len(ret.Items)
+	}, 30*time.Second, 1*time.Second).Should(BeZero())
+}
+
+func createNamespaces(argNamespaces ...string) {
+	for _, ns := range argNamespaces {
+		Expect(
+			namespaces.Create(ns, client.Client),
+		).ToNot(HaveOccurred())
+	}
+}
+
+func createPodsInNamespace(namespace string, addSCTPServer bool) (*corev1.Pod, *corev1.Pod, *corev1.Pod) {
+
+	podA := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podA, "pod", "a")
+	pods.RedefinePodWithNetwork(podA, TEST_NETWORK)
+	addNetcatContainers(podA, addSCTPServer)
+	_, err := pods.RedefineAsPrivileged(podA, "")
+	Expect(err).To(BeNil())
+	podA.ObjectMeta.GenerateName = "testpod-a-"
+	podA = createAndStartPod(podA)
+
+	podB := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podB, "pod", "b")
+	pods.RedefinePodWithNetwork(podB, TEST_NETWORK)
+	addNetcatContainers(podB, addSCTPServer)
+	_, err = pods.RedefineAsPrivileged(podB, "")
+	Expect(err).To(BeNil())
+	podB.ObjectMeta.GenerateName = "testpod-b-"
+	podB = createAndStartPod(podB)
+
+	podC := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podC, "pod", "c")
+	pods.RedefinePodWithNetwork(podC, TEST_NETWORK)
+	addNetcatContainers(podC, addSCTPServer)
+	_, err = pods.RedefineAsPrivileged(podC, "")
+	Expect(err).To(BeNil())
+	podC.ObjectMeta.GenerateName = "testpod-c-"
+	podC = createAndStartPod(podC)
+
+	return podA, podB, podC
+}
+
+func addNetcatContainers(pod *corev1.Pod, addSCTPServer bool) {
+
+	AddTCPNetcatServerToPod(pod, port5555)
+	AddUDPNetcatServerToPod(pod, port5555)
+
+	AddTCPNetcatServerToPod(pod, port6666)
+	AddUDPNetcatServerToPod(pod, port6666)
+
+	AddIPTableDebugContainer(pod)
+
+	if addSCTPServer {
+		AddSCTPNetcatServerToPod(pod, port5555)
+		AddSCTPNetcatServerToPod(pod, port6666)
+	}
+}
+
+func createAndStartPod(pod *corev1.Pod) *corev1.Pod {
```

Comment:
can you move this function to pods?

Reply:
Sure. also refactored `CreateDPDKWorkload(...)`